### ISSUE TITLE
Parameter wrapping incompatible with ktlint with long function parameter

### ISSUE THREAD

**Issue by bethcutler** [facebookincubator/ktfmt#245]
Current ktfmt formatting:
```kotlin
private fun testWrapping(
  testParameter:
    (
      longEnoughNameToForceMultilineParams: MyVeryLongObject,
      anotherVeryLongName: ALongObject) -> Unit
) {
  // foobar
}
```

What ktlint expects:
```kotlin
private fun testWrapping(
  testParameter: (
    longEnoughNameToForceMultilineParams: MyVeryLongObject,
    anotherVeryLongName: ALongObject
  ) -> Unit
) {
  // foobar
}
```

I think the ktlint formatting looks better.  The ) on a new line would be standard for "google-style".  The ( *not* being on its own line is a bit more compact and it's more obvious what the parens are "attached" to.  However, the current style is similar to the version with a scoping function, e.g.: (Acceptable for both ktfmt and ktlint)
```kotlin
private fun testWrapping(
  testParameter:
    suspend (
      longEnoughNameToForceMultilineParams: MyVeryLongObject,
      anotherVeryLongName: ALongObject,
    ) -> Unit
) {
  // foobar
}
```

**Comment by bethcutler** [comment]
Similar issue:

Original (and ktlint's version, and my personal preference):
```kotlin
typealias Listener = (
  aVeryLongFoo: String,
  anotherVeryLongFoo: String,
  aThirdVeryLongFoo: String,
  aFourthVerylongFoo: String
) -> Unit
```

ktfmt's version (google-style, FWIW):
```kotlin
typealias Listener =
  (
    aVeryLongFoo: String,
    anotherVeryLongFoo: String,
    aThirdVeryLongFoo: String,
    aFourthVerylongFoo: String) -> Unit
```

**Comment by davidtorosyan** [comment]
Fixed by 1bebf0bc4677832a2b7004acf42a938baecd565a, closing.
